### ISSUE TITLE
[Feature Request ] add command for a file summary

### ISSUE THREAD

**Issue by Mannshoch** [IfcOpenShell/IfcOpenShell#2183]
It would be interesting to have some status information about the IFC:
`.\IfcConvert.exe --summary <file>`

- Count of unique bodies
- Count of all bodies
- type of bodies inside this file (CSG-primitive, Extrusion, BREP, Tesselation )
- File header properties

**Comment by Moult** [comment]
@Mannshoch have you considered using ifcopenshell-python for this? It sounds a little bespoke and arbitrary, and better suited perhaps to a custom script?

**Comment by Mannshoch** [comment]
No, I was not aware of it. I also never used Python until now. I don't know If Script is better than .exe. Currently I plan to convert ifc files by using the command by hand. May in a later case I may create a MS batch script to convert a whole directory. 

At moment I resceive several IFC4 but my CAD do only support ifc3x2 and because IfcOpenShell do not convert from 4 to 3x2 (and my CAD do poorly support ifc 2x3) I have to convert them into STEP.

**Comment by Moult** [comment]
There is a basic IFC4->IFC2X3 converter written but likely is lacking some features to convert geometry. If you help with testing it we can perhaps build the features to make it more functional?

It sounds for your usecase a script will be more appropriate given the amount of customisation you might need.

**Comment by Mannshoch** [comment]
Really Important is none of the Information. It's helping check the Quality of data (body type)  and comunicating with customers (IFC Version). 

If you need I could help testing it. But only if you could provide an .exe. I'm a bit limittet how I could use my PC here.

**Comment by BIMvoice** [comment]
Hi @Mannshoch, following up on this old request. Everything asked for here is available today via ifcopenshell-python, no bespoke command needed:

```python
import ifcopenshell
from collections import Counter

f = ifcopenshell.open("model.ifc")
print("Schema:", f.schema)
print("Header:", f.wrapped_data.header())
print("Entity counts:", Counter(e.is_a() for e in f))
print("Body types:", Counter(r.RepresentationType for r in f.by_type("IfcShapeRepresentation")))
```

For header info specifically without loading the whole model, there is also `ifcopenshell.util.file.IfcHeaderExtractor`, which Bonsai's Project tab already uses to show schema/description/timestamp in its UI. For the "Errors" part of the ask, `ifcopenshell.validate` covers schema/standard compliance.

Since this matches Dion's original suggestion of a custom script rather than a dedicated CLI flag, leaving it to a maintainer whether to close this or keep it open as a documentation/recipe request.
